### PR TITLE
Include runtime and workflow outputs in syntax highlighter

### PR DESCRIPTION
**Pull request by scottfrazer**
* Wire through the `meta` and `parameter_meta` sections into the `Task` class
* Add syntax highlighting / code fromatting for workflow outputs, runtime section, meta section, parameter_meta section

### CODE REVIEW

**Comment by scottfrazer** [comment]
WHEEL: @Horneth @kshakir

File: src/test/scala/wdl4s/SyntaxHighlightSpec.scala [hunk 1]
```diff
@@ -5,6 +5,45 @@ import org.scalatest.{Matchers, WordSpecLike}
 
 class SyntaxHighlightSpec extends Matchers with WordSpecLike {
 
+  "foo" should {
+    "bar" in {
```
Comment (Horneth) [inline src/test/scala/wdl4s/SyntaxHighlightSpec.scala]: are `foo` `bar` meant to be replaced ?

Comment (scottfrazer) [inline src/test/scala/wdl4s/SyntaxHighlightSpec.scala]: oops.  This whole block shouldn't exist anymore.  Silly me.

File: src/main/scala/wdl4s/formatter/SyntaxFormatter.scala [hunk 2]
```diff
@@ -98,6 +99,29 @@ class SyntaxFormatter(highlighter: SyntaxHighlighter = NullSyntaxHighlighter) {
     header
   }
 
+  private def formatRuntimeSection(task: Task, level: Int): String = {
+    task.runtimeAttributes.attrs match {
+      case m: Map[String, Seq[String]] if m.nonEmpty =>
```
Comment (geoffjentry) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: Could you do an if nonEmpty?

File: src/main/scala/wdl4s/formatter/SyntaxFormatter.scala [hunk 3]
```diff
@@ -98,6 +99,29 @@ class SyntaxFormatter(highlighter: SyntaxHighlighter = NullSyntaxHighlighter) {
...
+    }
+  }
+
+  private def formatRuntimeAttribute(key: String, value: Seq[String], level: Int) = {
+    val rhs = value match {
+      case s: Seq[String] if s.isEmpty => "\"\""
+      case s: Seq[String] if s.size == 1 => "\"" + s.head + "\""
+      case s: Seq[String] =>
+        val stringList = s.map(x => "\"" + x + "\"").mkString(", ")
+        s"[$stringList]"
```
Comment (kshakir) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: Use the three-args mkString: `s.mkString("[\"", "\", \"", "\"]")`

Comment (mcovarr) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: :sunglasses:

**Comment by kshakir** [comment]
:+1: after above comments

File: src/main/scala/wdl4s/formatter/SyntaxFormatter.scala [hunk 4]
```diff
@@ -123,12 +147,33 @@ class SyntaxFormatter(highlighter: SyntaxHighlighter = NullSyntaxHighlighter) {
...
         |}""".stripMargin
   }
 
+  private def formatWorkflowOutputs(outputs: Seq[WorkflowOutputDeclaration], level: Int): String = {
+    outputs match {
+      case x: Seq[WorkflowOutputDeclaration] if x.nonEmpty =>
+        val outputStrings = outputs.map(formatWorkflowOutput(_, 1))
+        indent(s"""${highlighter.keyword("output")} {
+                  |${outputStrings.mkString("\n")}
+                  |}""".stripMargin, 1)
```
Comment (Horneth) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: My understanding is that `1` should be `level` instead here ?

**Comment by Horneth** [comment]
:+1: after a couple nitpicks

File: src/main/scala/wdl4s/formatter/SyntaxFormatter.scala (hunk 2 above)
Comment (scottfrazer) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: I'm not sure I understand what you mean

Comment (geoffjentry) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: What else could this possibly match to? Isn't task.runtimeAttributes.attrs a Map[String, Seq[String]]?

File: src/main/scala/wdl4s/formatter/SyntaxFormatter.scala (hunk 4 above)
Comment (scottfrazer) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: yes indeed

File: src/main/scala/wdl4s/formatter/SyntaxFormatter.scala (hunk 2 above)
Comment (scottfrazer) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: you mean `case m: m.nonEmpty`?

Comment (geoffjentry) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: Sure, but my original question was why not just if/else on the nonEmpty. Although I suppose then you'd have to either pre-assign attrs or give the full reference twice. 

Either way I found the type ascription to be kinda confusing.

Comment (scottfrazer) [inline src/main/scala/wdl4s/formatter/SyntaxFormatter.scala]: Okay, I removed the type ascription

**Pull request merged**